API/Controllers/TournamentsController.cs

```diff
@@ -207,4 +207,83 @@ public async Task<IActionResult> DeleteAsync(int id)
         await tournamentsService.DeleteAsync(id);
         return NoContent();
     }
+
+    /// <summary>
+    /// Get all beatmaps which are pooled by this tournament
+    /// </summary>
+    /// <param name="id">Tournament id</param>
+    /// <returns>A collection of all pooled beatmaps</returns>
+    [HttpGet("{id:int}/beatmaps")]
+    [Authorize(Roles = $"{OtrClaims.Roles.User}, {OtrClaims.Roles.Client}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetBeatmapsAsync(int id)
+    {
+        TournamentDTO? result = await tournamentsService.GetAsync(id);
+        if (result is null)
+        {
+            return NotFound();
+        }
+
+        return Ok(await tournamentsService.GetPooledBeatmapsAsync(id));
+    }
+
+    /// <summary>
+    /// Adds beatmaps to a tournament by osu! id
+    /// </summary>
+    /// <param name="id">Tournament id</param>
+    /// <param name="osuBeatmapIds">A collection of osu! beatmap ids</param>
+    /// <returns>The tournament's collection of pooled beatmaps</returns>
+    /// <response code="404">The tournament does not exist</response>
+    /// <response code="200">The beatmaps were added successfully</response>
+    [HttpPost("{id:int}/beatmaps")]
+    [Authorize(Roles = OtrClaims.Roles.Admin)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> InsertBeatmapsAsync(int id, [FromBody] ICollection<long> osuBeatmapIds)
```

Comment:
I think it would be more appropriate to move this endpoint to the `TournamentsController.Admin.cs` partial?